.github/workflows/ci-precompiled.yaml

```diff
@@ -0,0 +1,177 @@
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Pre-Compiled End-to-end tests
+
+on:
+  schedule:
+    - cron: '30 22 * * *'
+
+jobs:
+  e2e-driver-version-compare:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix_values_not_empty: ${{ steps.set_kernel_version.outputs.matrix_values_not_empty }}
+      matrix_values: ${{ steps.set_kernel_version.outputs.matrix_values }}
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v4
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Set kernel version
+      id: set_kernel_version
+      env:
+        BASE_TARGET: "jammy"
+        DIST: "ubuntu22.04"
+      run: |
+        export PRIVATE_REGISTRY="ghcr.io"
+        DRIVER_VERSIONS=$(grep '^DRIVER_VERSIONS ?=' versions.mk | awk -F' ?= ' '{print $2}')
+        # currently azure image upgrade is failing
+        # KERNEL_FLAVORS=("aws" "azure" "generic" "nvidia" "oracle")
+        KERNEL_FLAVORS=("aws" "generic" "nvidia" "oracle")
+        echo "matrix_values_not_empty=0" >> $GITHUB_OUTPUT
+        kernel_versions=()
+        for KERNEL_FLAVOR in "${KERNEL_FLAVORS[@]}"; do
+          for driver_version in ${DRIVER_VERSIONS}; do
+            DRIVER_VERSION=$(echo "${driver_version}" | cut -d '.' -f 1)
+            source ./tests/scripts/findkernelversion.sh "$BASE_TARGET" "${KERNEL_FLAVOR}" "$DRIVER_VERSION" "$DIST"
```

Comment:
task runs daily , if there is any change in kernel version and driver version kernel mismatch sets: 
   matrix_values_not_empty 
    matrix_values